products/kitty-render.js

```diff
@@ -43,10 +37,42 @@ export function renderKittens(kitten) {
 
     const button = document.createElement('button');
     button.textContent = 'Add to cart';
-    li.appendChild(button);
+    
+    button.addEventListener('click', () => {
+        const cart = getFromLocalStorage('CART') || [];  
+
+        const itemInCart = findById(cart, kitten.id);
+
+        if (itemInCart === undefined) {
+
+            const newCartItem = {
+                id: kitten.id,
+                quantity: 1
+            };
 
+            cart.push(newCartItem);
+        } else {
+            itemInCart.quantity++;
+        }
+
+        setInLocalStorage('CART', cart);
+    });
     
+    li.appendChild(button);
 
     return li;
 
-}
\ No newline at end of file
+}    
+    
+export function getFromLocalStorage(key) {
+    const retrievedKey = localStorage.getItem(key);
+    return JSON.parse(retrievedKey);
+}
+    
+   
+export function setInLocalStorage(key, value) {
+    const stringifiedItem = JSON.stringify(value);
+    localStorage.setItem(key, stringifiedItem);   
+}
```

Comment:
functions need to return something, you probably want: 
return value; 
on line 75 1/2.